starters/vue3-apollo-quasar/src/components/NumberCounter/NumberCounter.vue

```diff
@@ -36,15 +36,42 @@
 </template>
 
 <script lang="ts">
-import { defineComponent } from 'vue';
+import { computed, defineComponent } from 'vue';
 
 export default defineComponent({
   name: 'NumberCounter',
 });
 </script>
 
 <script lang="ts" setup>
-import { useCounterStore } from 'src/stores';
+import { counts } from 'src/variables/counts';
+import { useQuery } from '@vue/apollo-composable';
+import gql from 'graphql-tag';
 
-const counter = useCounterStore();
+const COUNTERS_QUERY = gql`
+  query Counter {
+    count @client
+  }
+`;
+
+const { result, loading } = useQuery(COUNTERS_QUERY);
+
+const count = computed(() => (result.value ? result.value.count : 0));
+
+const updateCount = (countValue: number) => {
```

Comment:
This is counter-intuitive.
Rather than testing if `updateCount` does addition, subtraction and reset, create individual functions for `increment` `decrement` and `reset`

Also, the use of switch in this case would be unnecessary. Needs refactoring.

Reply:
Copy that